tests/test_util.py

```diff
@@ -54,6 +61,43 @@ def test_with_statement(self):
         self.assertEqual(l, [1, 2, 3, 4])
 
 
+class TestProblemCache(unittest.TestCase):
+    def setUp(self):
+        self.reference = TESTMODEL.solve().fluxes
+        self.n_constraints = len(TESTMODEL.solver.constraints)
+        self.n_variables = len(TESTMODEL.solver.variables)
+
+    def test_cache_problem(self):
+        lmoma(model=TESTMODEL, reference=self.reference)
+
+        # After running lmoma without cache, the number of variables and constraints remains the same
+        self.assertEqual(self.n_constraints, len(TESTMODEL.solver.constraints))
+        self.assertEqual(self.n_variables, len(TESTMODEL.solver.variables))
+
+        cache = ProblemCache(TESTMODEL)
+        lmoma(model=TESTMODEL, reference=self.reference, cache=cache)
+        # After running lmoma with cache, the number of variables is 2 times the original number
+        self.assertEqual(2*self.n_variables, len(TESTMODEL.solver.variables))
```

Comment:
I agree with you! However, for practical reasons if lmoma fails it will fail on the unit tests for simulation methods. Is a necessary thing or more like an issue that can be fixed later?